apps/aeutils/src/aeu_requests.erl

```diff
@@ -42,16 +42,20 @@ ping(Peer) ->
             {error, Reason};
         {ok, Map} ->
             lager:debug("ping response (~p): ~p", [Uri, pp(Map)]),
-            aec_sync:compare_ping_objects(PingObj, Map),
-            check_returned_source(Map, list_to_binary(aec_peers:uri(Peer))),
-            {ok, Map};
+            case aec_sync:compare_ping_objects(PingObj, Map) of
+                ok ->  
+                    check_returned_source(Map, list_to_binary(aec_peers:uri(Peer))),
+                    {ok, Map};
+                Error ->
```

Comment:
Match against `{error, _}`, nice to have.

Reply:
That's really a matter of taste. We have dialyzer to verify that indeed the Error always is {error, _} and therefore don't need to write it explicitely.